plugin/pkg/scheduler/core/generic_scheduler.go

```diff
@@ -422,11 +473,339 @@ func EqualPriorityMap(_ *v1.Pod, _ interface{}, nodeInfo *schedulercache.NodeInf
+		}
+		for _, failedPred := range failedPredicates {
+			if failedPred != predicates.ErrPodAffinityNotMatch {
+				return nil, false
+			}
+		}
+		// If we reach here, it means that the pod cannot be scheduled due to pod
+		// affinity or anti-affinity. Since failure reason for both affinity and
+		// anti-affinity is the same, we cannot say which one caused it. So, we try
+		// adding pods one at a time and see if any of them satisfies the affinity rules.
+		for i, p := range potentialVictims.Items {
+			existingPod := p.(*v1.Pod)
+			addPod(existingPod)
+			if fits, _, _ = podFitsOnNode(pod, meta, nodeInfoCopy, fitPredicates, nil); !fits {
+				removePod(existingPod)
+			} else {
+				// We found the pod needed to satisfy pod affinity. Let's remove it from
+				// potential victims list.
+				// NOTE: We assume that pod affinity can be satisfied by only one pod,
+				// not multiple pods. This is how scheduler works today.
+				potentialVictims.Items = append(potentialVictims.Items[:i], potentialVictims.Items[i+1:]...)
+				break
+			}
+		}
+		if !fits {
+			return nil, false
+		}
+	}
+	victims := []*v1.Pod{}
+	// Try to reprieve as many pods as possible starting from the highest priority one.
+	for _, p := range potentialVictims.Items {
+		lpp := p.(*v1.Pod)
+		addPod(lpp)
+		if fits, _, _ := podFitsOnNode(pod, meta, nodeInfoCopy, fitPredicates, nil); !fits {
+			removePod(lpp)
+			victims = append(victims, lpp)
+			glog.V(5).Infof("Pod %v is a potential preemption victim on node %v.", lpp.Name, nodeInfo.Node().Name)
+		}
+	}
+	return victims, true
+}
+
+// nodesWherePreemptionMightHelp returns a list of nodes with failed predicates
+// that may be satisfied by removing pods from the node.
+func nodesWherePreemptionMightHelp(pod *v1.Pod, nodes []*v1.Node, failedPredicatesMap FailedPredicateMap) map[string]bool {
+	potentialNodes := map[string]bool{}
+	for _, node := range nodes {
+		unresolvableReasonExist := false
+		failedPredicates, found := failedPredicatesMap[node.Name]
+		// If we assume that scheduler looks at all nodes and populates the failedPredicateMap
+		// (which is the case today), the !found case should never happen, but we'd prefer
+		// to rely less on such assumptions in the code when checking does not impose
+		// significant overhead.
+		if found {
+			for _, failedPredicate := range failedPredicates {
+				switch failedPredicate {
+				case
+					predicates.ErrNodeSelectorNotMatch,
+					predicates.ErrPodNotMatchHostName,
+					predicates.ErrTaintsTolerationsNotMatch,
+					predicates.ErrNodeLabelPresenceViolated,
+					predicates.ErrNodeNotReady,
+					predicates.ErrNodeNetworkUnavailable,
+					predicates.ErrNodeUnschedulable,
+					predicates.ErrNodeUnknownCondition:
+					unresolvableReasonExist = true
+					break
+				case predicates.ErrPodAffinityNotMatch:
+					// if pod didn't schedule due to pod affinity/anti-affinity and it doesn't
+					// have any anti-affinity rule, it has failed due to pod affinity. So,
```

Comment:
I think this is also not true - it may be anti-affinity rule of one of the already existing pods..

@davidopp - am I missing something?

Reply:
Yes what you wrote is correct. However, the conservative thing to do in this function is to decide unresolvableReasonExist = false since that will cause you to try preemption. I believe that means that the issue you mentioned will just cause extra work and not a correctness problem. @bsalamat please verify.

Reply:
You're right, sorry about that!